static/js/stream_muting.js

```diff
@@ -67,4 +68,24 @@ export function update_is_muted(sub, value) {
         )}'] #sub_is_muted_setting .sub_setting_control`,
     );
     is_muted_checkbox.prop("checked", value);
+
+    $(".mute-note").toggleClass("hide-mute-note", !value);
+    $(".sub_notification_setting").toggleClass("muted-sub", value);
+    $(".sub_notification_setting").find("input[type='checkbox']").prop("disabled", value);
+
+    const row = $(`.stream-row[data-stream-id='${CSS.escape(sub.stream_id)}']`);
+    update_ui_muted_stream_checkboxes(row, value);
+}
+
+export function update_ui_muted_stream_checkboxes(row, value) {
+    $(row).toggleClass("control-label-disabled", value);
+    if (value) {
+        $(row).find(".unmute_stream").show();
+    } else {
+        $(row).find(".unmute_stream").hide();
+    }
+    $(row).find("input").prop("disabled", value);
+    $(row)
+        .find('[name="push_notifications"]')
+        .prop("disabled", !page_params.realm_push_notifications_enabled || value);
```

Comment:
It would be nice to not need to have this logic here; is there a way we can share the code for computing whether this should be marked as disabled with the existing rendering code?

One theory would be to replace this logic with just rerendering and replacing the whole row from scratch -- often that leads to more robust code.

Reply:
I tried this approach originally, but the problem with this is that either the newly rendered row is placed at the end of the table, even if it wasn't before (which looks weird), or that you have to re render all the rows and somehow figure out the correct order and render them in this order, which seemed like unnecessary overhead to me.

Reply:
OK, we can maybe leave this for a follow-up issue idea. I'd expect it to be possible with a library list `list_render` but not something we want to implement manually.